tiled/client/sync.py

```diff
@@ -0,0 +1,112 @@
+import collections.abc
+import itertools
+
+from ..structures.core import StructureFamily
+from ..structures.data_source import DataSource, Management
+from .base import BaseClient
+
+
+def sync(
+    source: BaseClient,
+    dest: BaseClient,
+):
+    """
+    Copy data from one Tiled instance to another.
+
+    Parameters
+    ----------
+    source : tiled node
+    dest : tiled node
+
+    Examples
+    --------
+
+    Connect to two instances and copy data.
+
+    >>> from tiled.client import from_uri
+    >>> a = from_uri("http://localhost:8000", api_key="secret")
+    >>> b = from_uri("http://localhost:9000", api_key="secret")
+    >>> sync(a, b)
+
+
+    Copy select data.
+
+    >>> sync(a.items().head(), b)
+    >>> sync(a.search(...), b)
+
+    """
+    if hasattr(source, "structure_family"):
+        # looks like a client object
+        _DISPATCH[source.structure_family](source.include_data_sources(), dest)
+    elif isinstance(source, list):
```

Comment:
I feel like this condition could be simplified a little (albeit it might not work with awkward and sparse...):
```
if hasattr(source, "structure_family"):
   _DISPATCH[source.structure_family](source.include_data_sources(), dest)
else:
   _DISPATCH[StructureFamily.container](dict(source), dest)
```

Reply:
Good call!